apps/api/src/models/collectible-auction.model.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  CollectibleAuctionSchema,
+  CollectibleAuctionStatus,
+} from '@algomart/schemas'
+import { Model } from 'objection'
+
+import { AlgorandTransactionModel } from './algorand-transaction.model'
+import { BaseModel } from './base.model'
+import { CollectibleModel } from './collectible.model'
+import { CollectibleAuctionBidModel } from './collectible-auction-bid.model'
+import { UserAccountModel } from './user-account.model'
+
+export class CollectibleAuctionModel extends BaseModel {
+  static tableName = 'CollectibleAuction'
+  static jsonSchema = CollectibleAuctionSchema
+
+  appId!: number
+  collectibleId!: string
+  endAt!: string
+  reservePrice!: number
+  startAt!: string
+  status!: CollectibleAuctionStatus
+  transactionId!: string
+  userAccountId!: string
+
+  bids?: CollectibleAuctionBidModel[]
+  collectible?: CollectibleModel
+  transaction?: AlgorandTransactionModel
+  userAccount?: UserAccountModel
```

Comment:
Should the option to allow bid expiration be here, so that the next user would be awarded the collectible if it matched the reserve price, if the first bid payment failed? Wondering if the functionality should be automatic, or if it should be an option for the user, and/or if perhaps the user may want to accept/deny at the end of an auction instead of an automatic process. We could always add down the road though.

Reply:
Oh right, awesome!